forum/users/serializers.py

```diff
@@ -0,0 +1,8 @@
+from rest_framework_simplejwt.serializers import TokenObtainPairSerializer
+
+class CustomTokenObtainPairSerializer(TokenObtainPairSerializer):
+    @classmethod
+    def get_token(cls, user, role='unassigned'):
```

Comment:
Issues:
- `role='unassigned'`:
  - This hardcodes the role to 'unassigned' unless explicitly passed.
  - It completely ignores the actual role field from the user model, which defeats the purpose of assigning dynamic roles.
- Not pulling the role from the user object:
  - The role should reflect the actual value stored in the database for the authenticated user.

Reply:
Done. Now get_token uses enums; but again - I assume that due to authentication flow the responsibility to give certain role (investor/startup) when he enters profile is on the profile endpoints views.

Reply:
Given this, maintaining the role selection at the token level during login seems like the right approach. It allows the role to be explicitly set for the session without directly relying on the database field, which reflects the user's general roles rather than the session-specific one. This distinction is essential for the flow you’ve described.
One thing we could consider is ensuring a robust mechanism for validating the selected role during login. For example:
- Verify that the selected role aligns with one of the roles stored in the database for the user.
- Log the role-switching activity for better traceability.